lib/manageiq/messaging/client.rb

```diff
@@ -2,31 +2,105 @@ module ManageIQ
   module Messaging
     class Client
       def self.open(options)
+        # TODO: replace with a AMQP client
         StompClient.new(options)
       end
 
-      def publish_background_job(options)
-        BackgroundJob.push(self, options)
-      end
-
-      def subscribe_background_job(options)
-        BackgroundJob.subscribe(self, options)
+      # options
+      #   :service    (service and affinity are used to determine the queue name)
+      #   :affinity   (optional)
+      #   :class_name (optional)
+      #   :message (e.g. method_name or message type)
+      #   :payload (user defined structure, following are some examples)
+      #     :instance_id
+      #     :args
+      #     :miq_callback
+      #   :sender    (optional, type of the publisher)
+      #   :sender_id (optional, identity of the publisher)
+      #   <other queue options TBA>
+      #
+      def publish_message(options)
+        Queue.publish(self, options)
       end
 
-      def publish_queue(options)
-        Queue.publish(self, options)
+      # An aggregate version of #publish_message but for better performance
```

Comment:
for future reference, not sure why someone would use this
Maybe we can put a little more into the comments (another PR)

Or not even a PR, but just comments around the group